service/controller/azuremachinepool/handler/nodepool/create_deployment_uninitialized.go

```diff
@@ -130,7 +130,10 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 				return currentState, microerror.Mask(err)
 			}
 
-			if desiredParameters.Scaling.MinReplicas == desiredParameters.Scaling.MaxReplicas && desiredParameters.Scaling.CurrentReplicas != int32(*vmss.Sku.Capacity) {
+			if desiredParameters.Scaling.MinReplicas == desiredParameters.Scaling.MaxReplicas &&
```

Comment:
Unrelated with this PR, but it panic'd during tests so I added a new check to avoid the panic

Reply:
weird